### DISCUSSION TITLE
Homeassistant Climate entity - help needed

### DISCUSSION THREAD

**Discussion by seaverd** [tagyoureit/nodejs-poolController#479] [Q&A]
I need some help with the Home Assistant Climate entity.  I have read through the homeassistant wiki on this GitHub project and am unable to get it to work.   I have created the input number entity in HA, and was able to create my own nodered flow to keep the input number in sync with the nodejsPC dashboard.  My issue is the climate entity is not interacting correctly.  I believe I have all topics modified for my install, and have the correct pool bodies referenced…pool controller is nixie shared body as I have a spa.  

Wondering if another homeassistant user could post their climate entity.

Here is mine that is not working:

`  - platform: mqtt
    name: "Pool Heater"
    unique_id: pool_heater
    min_temp: 55
    max_temp: 80
    modes:
      - "off"
      - "heat"
    current_temperature_topic: "nixie-sharedbody/state/temps/bodies/1/pool/temp"
    value_template: "{{ value_json.temp }}"
    mode_state_topic: "nixie-sharedbody/state/temps/bodies/1/pool/heatMode"
    mode_state_template: >-
      {% if value_json.heatMode.val == 0 %}
        off
      {% elif value_json.heatMode.val == 1 %}
        off
      {% elif value_json.heatMode.val == 2 %}
        heat
      {% endif %}
    mode_command_topic: "nixie-sharedbody/state/body/heatMode"
    mode_command_template: >-
      {% set values = { 'heat':'{"heatMode":2, "id":1}', 'off':'{"heatMode":1, "id":1}'} %}    
      {{ values[value] if value in values.keys() else '{"heatMode":1, "id":1}' }}
    action_topic: "nixie-sharedbody/state/temps/bodies/1/pool/heatStatus"
    action_template: >-
      {% if value_json.heatStatus.val == 0 %}
        off
      {% elif value_json.heatStatus.val == 1 %}
        heating
      {% endif %}
    temperature_command_topic: "pool/setpoint" #REQUIRES AN AUTOMATION - SEE BELOW
    temperature_state_topic: "nixie-sharedbody/state/temps/bodies/1/pool/setPoint"
    temperature_state_template: "{{ value_json.setPoint }}"
`

**Comment by Exit2Studios** [comment]
Before you go too far down this road, take a look at the upcoming breaking changes to mqtt in home assistant. They will no longer work with the “platform” method and will need to go under mqtt:

**Comment by Crewski** [comment]
I had to adjust the setup from the Wiki to work with my heater as it is a heat pump.  Maybe you need to adjust yours to work with your heater type?

Here is my config:

```
mqtt:
  climate:
    - name: "Pool Heater"
      min_temp: 60
      max_temp: 95
      modes:
        - "off"
        - "heat"
      current_temperature_topic: "pool/state/temps/bodies/1/pool/temp"
      value_template: "{{ value_json.temp }}"
      mode_state_topic: "pool/state/temps/bodies/1/pool/heatMode"
      mode_state_template: >-
        {% if value_json.heatMode.val == 0 %}
          off
        {% elif value_json.heatMode.val == 1 %}
          off
        {% elif value_json.heatMode.val == 9 %}
          heat
        {% endif %}
      mode_command_topic: "pool/state/body/heatMode"
      mode_command_template: >-
        {% set values = { 'heat':'{"heatMode":9, "id":1}', 'off':'{"heatMode":1, "id":1}'} %}    
        {{ values[value] if value in values.keys() else '{"heatMode":1, "id":1}' }}
      action_topic: "pool/state/temps/bodies/1/pool/heatStatus"
      action_template: >-
        {% if value_json.heatStatus.val == 0 %}
          off
        {% elif value_json.heatStatus.val == 4 %}
          heating
        {% endif %}
      temperature_command_topic: pool/setpoint #REQUIRES AN AUTOMATION - SEE BELOW
      temperature_state_topic: "pool/state/temps/bodies/1/pool/setPoint"
      temperature_state_template: "{{ value_json.setPoint }}"
```

In the mode_state_template I had to change the value_json.heatMode.val to a 9 to reflect the heat pump, same in the mode_command_template.  For the action_template I had to change value_json.heatStatus.val to 4.

Here are my mqtt message from when my heater is off and on.
![image](https://user-images.githubusercontent.com/8012153/172696956-dedf3f3d-36d4-4ef7-a170-d0e6173aad5d.png)

![image](https://user-images.githubusercontent.com/8012153/172696988-58cc5fef-d7f2-42a1-b626-88e79c5e6f8c.png)

**Reply by rstrouse** [reply]
Wanna bet that the HASS object doesn't understand hpheat.  The heater status is not that simple.  It can be heater, hpheat, mtheater, hybrid, dual, cooling, hpcool, solar, cooldown.  All of these indicate that the heater has some activity going on.

**Reply by Crewski** [reply]
@rstrouse mine is working properly as I only have "off" and "hpheat".  I was just sharing the changes I made to get it working.

@seaverd maybe you can provide what type of heater you have and some mqtt message of when it is on and off

**Comment by seaverd** [comment]
@rstrouse and @Crewski   Thank you so much for the information.   I see you
are using mqtt explorer...that is much better than what I was using and I
think that will help my troubleshooting.I am away and will not get back to
this on Monday.

Dan

**Reply by rstrouse** [reply]
Yeah MQTT Explorer is the way to go if you are ape-ing around with MQTT.
